### PR TITLE
reindex remote indices from the prod cluster into the rank cluster

### PR DESCRIPTION
**Pull request by harrisonpim**
No real need for full CCR, just do a [remote reindex](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-reindex.html#reindex-from-remote) instead

### CODE REVIEW

File: cli/commands/index.py [hunk 1]
```diff
@@ -184,6 +190,87 @@ def get(
...
+    pipeline_username = get_secret(secret_prefix + "es_username")
+    protocol = get_secret(secret_prefix + "protocol")
+    public_host = get_secret(secret_prefix + "public_host")
+    port = get_secret(secret_prefix + "port")
+    pipeline_host = f"{protocol}://{public_host}:{port}"
+
+    pipeline_client = elasticsearch.pipeline_client(
+        session=context.meta["session"],
+        pipeline_date=pipeline_date,
+    )
```
Comment (jamieparkinson) [inline cli/commands/index.py:241]: is there a reason we can't use `pipeline_client` in `services/elasticsearch.py` to do this?

File: cli/commands/index.py [hunk 2]
```diff
@@ -184,6 +190,87 @@ def get(
...
+        raise typer.BadParameter(f"{source_index} is a system index")
+
+    if dest_index is None:
+        dest_index = typer.prompt(
+            "What do you want to call the index in the rank cluster?",
+            default=source_index,
+        )
+    raise_if_index_already_exists(context=context, index=dest_index)
+
+    task = rank_client.reindex(
```
Comment (jamieparkinson) [inline cli/commands/index.py:257]: Is there any risk to the API cluster when we do this? I remember stuff going wrong in the past but can't remember the circumstances - might be worth having a warning "are you sure" sort of prompt.

**Review by jamieparkinson — APPROVED** [review]

File: cli/commands/index.py (hunk 1 above)
Comment (harrisonpim) [reply cli/commands/index.py:241]: The only reason for this big repeated block is that we need the host/credentials later on for

```json
"remote": {
    "host": pipeline_host,
    "username": pipeline_username,
    "password": pipeline_password,
},
```

and I don't think we can get those values back out of an existing client. It is ugly though...

Comment (jamieparkinson) [reply cli/commands/index.py:241]: In that case maybe create a `get_cluster_credentials` method to be used in both places? And accept that it'd be used twice when calling `pipeline_client` here as well

Comment (harrisonpim) [reply cli/commands/index.py]: I've morphed `get_secret()`  into `get_secrets()`, which achieves roughly the same thing. I think there's probably a cleverer version of this which uses the `context.meta` somehow, but this works for now 🤷

Let me know what you think

**Pull request merged**